### DISCUSSION TITLE
Which features are safe to enable on a low-memory device?

### DISCUSSION THREAD

**Discussion by arkanoid87** [bluenviron/mediamtx#1559] [Questions] [answered]
I'm using rtsp-simple-server on a low-memory linux device (256MB). It works nicely, but I occasionally I see kernel killing rtsp-simple-server due to OOM.

My source pipeline (gstreamer) use RTSP to publish stream(s).

I'd like to ask from a developer point of view which output stream/protocols/options are more memory hungry, and which are safe to use in a low-memory context and which are a no-go.

Thanks!

**Comment by aler9** [comment, marked as answer]
Hello, the feature that consumes most RAM is without doubt the HLS protocol, that generates video segments and stores them in the RAM. From my experience, a significant portion of the RAM is occupied by segments. You can store the segment on disk instead of on RAM by using the new `hlsDirectory` parameter and point to a temporary directory:

```yml
hlsDirectory: /tmp/mediamtx
```

**Reply by arkanoid87** [reply]
thanks!

[a comment by an automated account is not shown]
